src/containers/DataSet/index.js

```diff
@@ -0,0 +1,236 @@
+  s3_bucket,
+  s3_key,
+  ...props
+}) {
+  // 1. Check if the dataset is already processed, if true show a link to the download file
+  if (is_processed) {
+    if (is_available) {
+      return <DataSetReady {...props} />;
+    } else {
+      return <DataSetExpired />;
+    }
+  } else {
+    // 2. if it's not ready to be downloaded, then allow the user to set an email and receive an alert when its ready
+    if (!email_address) {
+      return <DatasetNoEmail {...props} />;
+    } else {
+      // 3. Allow the user to change its email if it's already added
+      return <DataSetWithEmail {...props} email={email_address} />;
+    }
+  }
+}
+
+/**
+ * This component gets rendereded in the DataSet page, when no email has been assigned
+ */
+function DatasetNoEmail({ id }) {
+  return (
+    <div>
+      <h1>
+        We’re putting your download file together. It usually takes 15- 20
+        minutes.
+      </h1>
+      <h2>
+        Enter your email and we will email you when the files are ready for
+        download.
+      </h2>
+
+      <EmailForm dataSetId={id} />
+
+      <div className="dataset__image">
+        <img src={ProcessingImage} alt="" />
+      </div>
+    </div>
+  );
+}
+
+/**
+ * In this state an email was already assigned to the Data Set
+ */
+class DataSetWithEmail extends React.Component {
+  state = {
+    emailUpdated: false
+  };
+
+  render() {
+    const { id, email } = this.props;
+
+    return (
+      <div>
+        <div className="dataset__way-container">
+          <section>
+            <h1>Your dataset is on the way!</h1>
+
+            {this.state.emailUpdated && (
+              <div className="color-success">
+                <i className="ion-checkmark-circled" /> Email updated
+              </div>
+            )}
+
+            <ModalManager
+              modalProps={{ className: 'dataset__email-modal', center: true }}
+              component={showModal => (
+                <p>
+                  We have sent a confirmation email to {email} (
+                  <Button onClick={() => showModal()} buttonStyle="link">
+                    change
+                  </Button>)
+                </p>
+              )}
+            >
+              {({ hideModal }) => (
+                <div>
+                  <h1 className="dataset__email-modal-title">Change Email</h1>
+                  <h4>Enter New Email</h4>
+                  <EmailForm
+                    email={email}
+                    dataSetId={id}
+                    onSubmit={() => {
+                      hideModal();
+                      this._setEmailUpdated();
+                    }}
+                  />
+                </div>
+              )}
+            </ModalManager>
+
+            <p>If you haven’t recevied it, please check your spam folders.</p>
+          </section>
+          <div className="dataset__way-image">
+            <img src={NextStepsImage} alt="" />
+          </div>
+        </div>
+        {false && (
```

Comment:
Hmm, I'm a little confused about this pattern. Won't this never render those lines of HTML?

Reply:
That's for the bottom part of the download page:

![image](https://user-images.githubusercontent.com/1882507/40669755-af9b9344-6335-11e8-9c72-e6d741d80c02.png)

Deepa, told me to hide those for now until those two pages are created. If you want we can remove them from the component

Reply:
ahh ok, can you just add a comment?

Reply:
sure, good idea :)